django/chat/tasks.py

```diff
@@ -65,11 +67,17 @@ def translate_file(file_path, out_message_id, target_language):
         # Wait for translation to finish
         result = poller.result()
 
-        # Estimate the cost
-        # usage = poller.details.total_characters_charged
-        # estimated_cost = calculate_cost(usage, settings.AZURE_DOCUMENT_TRANSLATION)
-
         out_message = Message.objects.get(id=out_message_id)
+        usage = poller.details.total_characters_charged
+        cost = Cost.objects.new(
+            user=out_message.chat.user,
+            feature="translate",
+            cost_type="translate-doc",
+            count=usage,
+        )
+        print(f"Cost: {cost}")
+        out_message.cost = float(out_message.cost) + float(cost.usd_cost)
```

Comment:
I also need to save the total costs inside the message, in order to display the cost per message to the user.
This might actually sum multiple cost objects, either of the same type (like translating multiple files, here) or of multiple types (embedding, in tokens, out tokens)